test/connect-timeout.js

```diff
@@ -1,37 +1,40 @@
 'use strict'
 
-const { test } = require('tap')
+const { tspl } = require('@matteo.collina/tspl')
+const { test, after } = require('node:test')
 const { Client, Pool, errors } = require('..')
 const net = require('node:net')
 const sleep = ms => Atomics.wait(new Int32Array(new SharedArrayBuffer(4)), 0, 0, Number(ms))
 
-test('prioritize socket errors over timeouts', (t) => {
-  t.plan(1)
+test('prioritize socket errors over timeouts', async (t) => {
+  t = tspl(t, { plan: 1 })
   const connectTimeout = 1000
   const client = new Pool('http://foobar.bar:1234', { connectTimeout: 2 })
 
   client.request({ method: 'GET', path: '/foobar' })
     .then(() => t.fail())
     .catch((err) => {
-      t.equal(err.code, 'ENOTFOUND')
+      t.strictEqual(err.code, 'ENOTFOUND')
```

Comment:
Results in UND_ERR_CONNECT_TIMEOUT. 
I guess, this is because they are in the same loop as the sleep call, so it is not enforcing the ENOTFOUND error before the socket timeout.